### PR TITLE
EDGINREACH-10: Add proxy controller, feignClient to communicate with …

### PR DESCRIPTION
**Pull request by andrew-khodyka**
## Purpose
Add proxy controller, feignClient to communicate with mod-inn-reach

## Approach

- Add RestController that handles all requests starting with "/innreach/v2"
- Add FeignClient, Configuration

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [ ] Duplications on new code is 3% or less
   - [ ] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/edge/EdgeInnReachApplicationTest.java [hunk 1]
```diff
@@ -1,9 +1,7 @@
 package org.folio.edge;
 
 import org.junit.jupiter.api.Test;
-import org.springframework.boot.test.context.SpringBootTest;
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 class EdgeInnReachApplicationTest {
```
Comment (dmtkachenko) [inline src/test/java/org/folio/edge/EdgeInnReachApplicationTest.java:5]: What is tested with this test?

File: src/main/java/org/folio/edge/domain/InnReachRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import org.folio.edge.domain.dto.InnReachRequest;
+import org.folio.edge.domain.exception.EdgeServiceException;
+
+@Log4j2
+@Component
+public class InnReachRequestBuilder {
```
Comment (dmtkachenko) [inline src/main/java/org/folio/edge/domain/InnReachRequestBuilder.java:17]: Any tests for this class?

Comment (andrew-khodyka) [reply src/main/java/org/folio/edge/domain/InnReachRequestBuilder.java:17]: Added

File: src/main/java/org/folio/edge/client/InnReachClientProxy.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSocketFactory;
+
+import feign.Client;
+import feign.Request;
+import feign.Response;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
```
Comment (oliinyko) [inline src/main/java/org/folio/edge/client/InnReachClientProxy.java:13]: Shoudn't it be @Log4j2 ?

File: src/main/java/org/folio/edge/controller/exception/ExceptionHandlerController.java [hunk 4]
```diff
@@ -52,7 +54,7 @@ public Error handleMissingHeaderException(MissingRequestHeaderException e) {
 
   @ExceptionHandler(BadCredentialsException.class)
   @ResponseStatus(HttpStatus.UNAUTHORIZED)
-  public Error handleBadCredentialsException(BadCredentialsException e){
+  public Error handleBadCredentialsException(BadCredentialsException e) {
```
Comment (oliinyko) [inline src/main/java/org/folio/edge/controller/exception/ExceptionHandlerController.java:57]: I think we should return api errors in format expected by central server
. You can check the schema here https://raw.githubusercontent.com/folio-org/mod-inn-reach/master/src/main/resources/swagger.api/schemas/d2ir/innReachResponseDTO.json

Comment (andrew-khodyka) [reply src/main/java/org/folio/edge/controller/exception/ExceptionHandlerController.java:57]: I checked documentation, it seems we have two types of error responces.
The first one implemented now is for authentication errors.
The second one you listed is for common mistakes.
I think we need to implement the second option as well, as well as refactor the code to use the appropriate error response for different cases. I would create a separate task for this.

File: src/main/java/org/folio/edge/client/InnReachClientProxy.java (hunk 3 above)
Comment (andrew-khodyka) [reply src/main/java/org/folio/edge/client/InnReachClientProxy.java:13]: Yes again) forgot the replace it, thanks!

**Review by dmtkachenko — APPROVED** [review]

**Review by oliinyko — APPROVED** [review]

**Pull request merged**